webviz_subsurface/plugins/_co2_leakage/_utilities/surface_publishing.py

```diff
@@ -0,0 +1,104 @@
+from dataclasses import dataclass
+from typing import List, Union, Optional
+
+import xtgeo
+
+from webviz_subsurface._providers import (
+    SurfaceAddress,
+    SurfaceServer,
+    EnsembleSurfaceProvider,
+    QualifiedSurfaceAddress,
+    StatisticalSurfaceAddress,
+    SimulatedSurfaceAddress,
+)
+from webviz_subsurface._providers.ensemble_surface_provider.ensemble_surface_provider import \
+    SurfaceStatistic
+from webviz_subsurface.plugins._co2_leakage._utilities.plume_extent import truncate_surfaces
+
+
+@dataclass
+class TruncatedSurfaceAddress:
```

Comment:
This is interesting and should probably be seen in a larger context on how to handle calculated surfaces.
I guess it also raises the question if the current FMU standards for surface naming are "rich" enough.
Nice workaround :)

Reply:
Agree that this is probably a bigger-picture discussion. One way could be to extend `SurfaceServer` to handle qualified addresses in a best-effort manner (when the type is unknown), e.g. using a simpler hashing mechanism for those types of objects.

With regards to meta data, I guess there is some work going on in xtgeoapp_grd3dmaps and fmu-dataio that might relate to this, but I am not familiar with the details.